### PR TITLE
Add support for napt in place of kapt

### PR DESCRIPTION
**Pull request by ZacSweers**
This allows us to avoid kapt in more places

https://github.com/sergei-lapin/napt

### CODE REVIEW

File: slack-plugin/src/main/kotlin/slack/gradle/SlackExtension.kt [hunk 1]
```diff
@@ -56,28 +56,34 @@ public abstract class SlackExtension @Inject constructor(objects: ObjectFactory)
...
       val allowKsp = slackProperties.allowKsp
       val allowMoshiIr = slackProperties.allowMoshiIr
+      val allowNapt = slackProperties.allowNapt
 
       /** Marks this project as needing kapt code gen. */
       fun markKaptNeeded(source: String) {
-        kaptRequired = true
+        if (allowNapt) {
+          naptRequired = true
+          // Apply kapt for them
```
Comment (seankim-android) [inline slack-plugin/src/main/kotlin/slack/gradle/SlackExtension.kt:71]:
```suggestion
          // Apply napt for them
```

**Review by seankim-android — APPROVED** [review]
😮 https://github.com/sergei-lapin/napt says it increases annotation processing speed by ~ 50%. Nice!

**Comment by ZacSweers** [comment]
Will wait for https://github.com/sergei-lapin/napt/pull/7 to be released before merging this, as we'll wanna disable that by default

**Comment by ZacSweers** [comment]
Gonna merge with this as-is and update later, wanna debug https://github.com/sergei-lapin/napt/issues/8

**Pull request merged**